### ISSUE TITLE
Sorting Select Box on Shop Page is Not Accessible

### ISSUE THREAD

**Issue by BFTrick** [woothemes/woocommerce#44782]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When a user is on the Shop page there is the Sorting select box. There are several options that users can select from to sort the Shop page.

<img width="852" alt="Screenshot 2024-02-19 at 11 56 28 AM" src="https://github.com/woocommerce/woocommerce/assets/1065372/d747bcd3-3a18-420e-9581-ed023b7772d4">

Unfortunately, if you're using screen reading software there is no way to select an option from the select box, and then have your screen reading software narrate your choice to you. The page immediately refreshes.

### Expected behavior

There should be some ability to know what the options in the select box are before selecting one.

Or, you select a select box option, screen reading software narrates your choice, and then press a button for the page to reload.

---

The second option might be better. As a user who uses the keyboard it's not intuitive to have the page refresh when I click a dropdown.

### Actual behavior

Here's an example of what happens with Storefront.

As you will see, a user selects an option from the dropdown and the page immediately refreshes without even knowing what you selected.

https://github.com/woocommerce/woocommerce/assets/1065372/1c325ee8-f20c-4f5c-beba-4c790a1394a1

### Steps to reproduce

1. Install WooCommerce & a bunch of products (at least 20 to paginate)
2. Install Storefront & activate
3. Navigate to Shop page
4. Turn on screen reading software
5. Navigate by keyboard to the select box and choose an item

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): http://woo-core.local
Site address (URL): http://woo-core.local
WC Version: 8.5.2
REST API Version: ✔ 8.5.2
WC Blocks Version: ✔ 11.8.0-dev
Action Scheduler Version: ✔ 3.7.1
Log Directory Writable: ✔
WP Version: 6.4.3
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.16.0
PHP Version: 8.1.23
PHP Post Max Size: 1,000 MB
PHP Time Limit: 1200
PHP Max Input Vars: 4000
cURL Version: 8.4.0
(SecureTransport) LibreSSL/3.3.6

SUHOSIN Installed: –
MySQL Version: 8.0.16
Max Upload Size: 300 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 8.5.2
WC Database Prefix: wp_
Total Database Size: 3.89MB
Database Data Size: 2.20MB
Database Index Size: 1.69MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 1.06MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_posts: Data: 0.06MB + Index: 0.06MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.06MB + Index: 0.02MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_addresses: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_operational_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_orders: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_wc_orders_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 11
page: 7
post: 2
product: 18
revision: 3
wp_global_styles: 1
wp_navigation: 1
wp_template: 2

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (1) ###

WooCommerce: by Automattic – 8.5.2 (update to version 8.6.0 is available)

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to Woo.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature screen enabled: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### WC Pages ###

Shop base: #7 - /shop/
Cart: #8 - /cart/
Checkout: #9 - /checkout/
My account: #10 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 4.5.4
Author URL: https://woo.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
product-variation-management
product-virtual-downloadable
product-external-affiliate
product-grouped
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: customize-store
minified-js
new-product-management-experience
product-linked
settings
async-product-editor-category-field

Daily Cron: ✔ Next scheduled: 2024-02-19 21:22:38 +00:00
Options: ✔
Notes: 75
Onboarding: completed

### Action Scheduler ###

Complete: 32
Oldest: 2024-02-05 21:23:41 +0000
Newest: 2024-02-19 18:48:26 +0000

Failed: 1
Oldest: 2024-02-05 21:24:38 +0000
Newest: 2024-02-05 21:24:38 +0000

Pending: 1
Oldest: 2024-02-20 18:48:26 +0000
Newest: 2024-02-20 18:48:26 +0000

### Status report information ###

Generated at: 2024-02-19 18:58:55 +00:00
`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by amesplant** [comment]
I have started looking into this.

**Comment by amesplant** [comment]
@BFTrick I am looking into this issue and using VoiceOver as my screen reader, the selections are being announced correctly and thus am unable to reproduce this issue. For some reason my recorder did not capture the actual select menu opening up.
[Shop-Filter.webm](https://github.com/user-attachments/assets/1a924b7a-c498-442e-8661-63ae35261159)

**Comment by vikrampm1** [comment]
@BFTrick do you have any comments on the above? Moving this back to "To do" until then.

**Comment by vikrampm1** [comment]
@BFTrick this is currently blocked and still waiting on your feedback. If this is no longer an issue, then we can close it.

**Comment by amberhinds** [comment]
I think this is one we raised while auditing @BFTrick's company's website. I'll review and respond later in this week.

**Comment by amberhinds** [comment]
I retested this with both VoiceOver in Chrome and NVDA in Chrome and Firefox. I identified a different issue with the combox (which I will flag separately in our audit report for Woo), however I cannot replicate the issue where you can use the arrow keys to navigate the combo box and it does not announce the options. 

I suspect that this was fixed in some other way since the issue was originally opened in February and this issue can safely be closed.

**Comment by Manussakis** [comment]
Thanks @amberhinds !

@vikrampm1 This issue can be closed. I am moving it to Woo review.

**Comment by jeffpaul** [comment]
@jamesckemp FYI it looks like this can be closed

**Comment by vikrampm1** [comment]
@woocommerce/woo-fse as noted above can someone please help close this issue? Thanks.

**Comment by kmanijak** [comment]
Closing the issue as per comments above, thanks!
